scripts/collections-manager

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python
+import glob
+import os
+import subprocess
+import sys
+import tempfile
+import time
+from contextlib import contextmanager
+from importlib import import_module
+from pathlib import Path
+from typing import Generator, List
+
+import typer
+import yaml
+
+DEFAULT_ROOT = Path("~/src/github.com/PrefectHQ").expanduser()
+
+app = typer.Typer(help="Utilities for managing Prefect collections en masse")
+
+
+def prefect_collections() -> List[str]:
+    entries = []
+
+    for file in glob.glob("docs/integrations/catalog/*.yaml"):
+        if file.lower().endswith("/template.yaml"):
+            continue
+
+        with open(file) as f:
+            entries.append(yaml.safe_load(f))
+
+    entries = [e for e in entries if e["author"] == "Prefect"]
+    return [e["collectionName"] for e in entries]
+
+
+def ensure_repo(root: Path, collection: str) -> Path:
+    repo = root / collection
+    if not (repo / ".git").exists():
+        subprocess.check_call(
+            [
+                "git",
+                "clone",
+                f"git@github.com:PrefectHQ/{collection}.git",
+                str(repo),
+            ]
+        )
+
+    return repo
+
+
+def repositories(root: Path) -> Generator[Path, None, None]:
+    if not root.exists():
+        raise typer.BadParameter(f"Root directory {root} does not exist")
+
+    for collection in prefect_collections():
+        yield ensure_repo(DEFAULT_ROOT, collection)
+
+
+def _shell(repo: Path, command: str):
+    env = os.environ.copy()
+    env["COLLECTION"] = repo.name
+    subprocess.check_call(command, shell=True, cwd=str(repo))
+
+
+def _in_env(repo: Path, command: str):
+    env = os.environ.copy()
+    env["COLLECTION"] = repo.name
+    env["PYENV_VERSION"] = repo.name
+    env["PYENV_DIR"] = str(repo)
+    subprocess.check_call(f"pyenv exec {command}", shell=True, cwd=str(repo), env=env)
+
+
+@app.command()
+def initialize_python_environment(python_version: str, root: Path = DEFAULT_ROOT):
+    """Sets up each repo's Python environment using pyenv and pip.  Expects that you
+    have pyenv configured locally already."""
+    for repo in repositories(root):
+        if not (repo / ".python-version").exists():
+            _shell(repo, f"pyenv virtualenv {python_version} {repo.name}")
+            _shell(repo, f"pyenv local {repo.name}")
+
+        _in_env(repo, "pip install -e '.[dev]'")
+        _in_env(repo, "pre-commit install")
```

Comment:
This is where someone could stitch in `conda` or `venv` support pretty easily, as long as we preserve the meaning of `_in_env`, this can work with any VE strategy